ecl/hqlcpp/hqlhtcpp.cpp

```diff
@@ -12975,8 +12982,16 @@ void HqlCppTranslator::doBuildAggregateProcessTransform(BuildCtx & ctx, BoundRow
             }
             break;
         }
+
+        if (isAggregateOp && isDynamicOffset)
+            throwError1(HQLERR_AggregateDyanmicOffset, str(target->queryChild(1)->queryId()));
+
         if (target->queryType()->getSize() == UNKNOWN_LENGTH)
+        {
            isVariableOffset = true;
+           if (isAggregateOp)
+               isDynamicOffset = true;
```

Comment:
Ah, perhaps you wrote
static bool reorderAggregateFields(HqlExprArray & aggregateFields, HqlExprArray & aggregateAssigns) 
first and copied it and added the throw?
(Though you would have needed to change "isVariableOffset = true" which is an else in reorderAggregateFields. Should the logic be identical between the two?)